bin/specextract

```diff
@@ -79,381 +81,449 @@ v4 = make_verbose_level(toolname, 4)
-    dmextract.opt = "pha1"
-    dmextract.clobber = clobber
-    dmextract.verbose = str(verbose)
+    v2("Searching for P2_RESP calibration file...")
 
-    if channel == "1:1024:1":
-        dmextract.infile = f"{infile}[bin {ptype}]"
-    else:
-        dmextract.infile = f"{infile}[bin {ptype}={channel}]"
+    calquiz_results = call_calquiz(infile,"SC_MATRIX",rmffile,caldb_p2_resp_file,verbose)
+    numCalFiles = calquiz_results[0]
 
-    if instrument == "ACIS":
-        dmextract.wmap = f"[energy={ewmap}][bin {binwmap}]"
-    else:
-        dmextract.wmap = ""
+    caldb_p2_resp_file = calquiz_results[1]
 
-    dmextract()
+    v3(f"Found the following P2_RESP file(s): {caldb_p2_resp_file}")
 
-    return specfile
+    if not numCalFiles or numCalFiles in [0.0, 0, 0.]:
+        
+        v1("Cannot use mkacisrmf because no P2_RESP files were found.\n")
 
+        v1(f"Please reprocess {infile} with acis_process_events if you wish to use mkacisrmf, unless the focal plane temperature is greater than -110C and the observation is taken in graded-mode or the extraction region is on a front-illuminated CCD.\n")
+
+        v1("Using mkrmf...\n")
+
+        rmftool = "mkrmf"
+
+    else:
+        
+        v1("Using mkacisrmf...\n")
+
+        rmftool = "mkacisrmf"
+
+    return rmftool
 
-##########################################################################
-# Usage:
-#   build_rmf_ext(rmftool, ptype, full_outroot, ebin,
-#              rmfbin, clobber, verbose, specfile, weightfile)
-#
-#
-# Aim:
-#   Run either mkrmf or mkacisrmf depending on input conditions.
-#   For mkrmf:  input CALDB and weight file and return RMF name.
-#   For mkacisrmf:  input CALDB and WMAP and return RMF name.
-#
-##########################################################################
 
-def build_rmf_ext(rmftool, ptype, full_outroot, ebin, rmfbin, clobber, verbose, specfile, weightfile, wmap_clip, wmap_sky2tdet, infostr):
 
+def build_rmf_ps(rmftool, evt_filename, infile, ptype, full_outroot,
+                 ebin, rmfbin, clobber, verbose, specfile,
+                 feffile, ccd_id, chipx, chipy, infostr):
+	
     """
-    RMF filename to return
+    Run either mkrmf or mkacisrmf depending on input conditions.
+    For mkrmf: input FEF file and no weights file and return RMF name.
+    For mkacisrmf: input CALDB and WMAP='none' and return RMF name.
+    
+    Returns unweighted ACIS RMF filename.
     """
 
     v1(infostr)
     
     rmffile = f"{full_outroot}.rmf"
 
-    if "mkrmf" == rmftool:
-
-        mkrmf.punlearn()
+    try:
+        with new_pfiles_environment(ardlib=True,copyuser=False):
+            if rmftool == "mkrmf":
+                mkrmf.infile = feffile
+                mkrmf.outfile = rmffile
+                mkrmf.logfile = ""
+                mkrmf.weights = ""
+                mkrmf.axis1 = f"energy={ebin}"
+                mkrmf.axis2 = rmfbin
+                mkrmf.clobber = clobber
+                mkrmf.verbose = str(verbose)
```

Comment:
we shouldn't need str(verbose), it should accept an integer (and I'd claim it's a bug if it doesn't),

Reply:
revised!